app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -331,6 +333,13 @@ private float calculateCompletionPercentage(int completedFields, int totals) {
                                     emptyMandatoryFields.put(fieldViewModel.uid(), fieldViewModel);
                                 }
                             }
+                            int lastIndex = fields.size() - 1;
+                            FieldViewModel field = fields.get(lastIndex);
+                            if (field instanceof EditTextViewModel &&
+                                    ((EditTextViewModel) field).valueType() != ValueType.LONG_TEXT
```

Comment:
can we have a section with empty fields? or with matrix and have fields size == 0?